### PR TITLE
doc: present how to create a vim-plugin overlay

### PR DESCRIPTION
**Pull request by teto**
it is possible to maintain an out of tree list of neovim plugins with this method

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: doc/languages-frameworks/vim.section.md [hunk 1]
```diff
@@ -154,6 +154,22 @@ in
 }
 ```
 
+You can use the updater script to generate basic packages out of a custom vim
+plugin list:
+`
+pkgs/misc/vim-plugins/update.py -i vim-plugin-names -o generated.nix --no-commit
+`
```
Comment (SuperSandro2000) [inline doc/languages-frameworks/vim.section.md:161]:
```suggestion
```
pkgs/misc/vim-plugins/update.py -i vim-plugin-names -o generated.nix --no-commit
```
```

File: doc/languages-frameworks/vim.section.md [hunk 2]
```diff
@@ -154,6 +154,22 @@ in
...
 
+You can use the updater script to generate basic packages out of a custom vim
+plugin list:
+```
+pkgs/misc/vim-plugins/update.py -i vim-plugin-names -o generated.nix --no-commit
+```
+with the content of vim-plugin-names be like:
+```
+pwntester/octo.nvim
+```
```
Comment (SuperSandro2000) [inline doc/languages-frameworks/vim.section.md:165]:
```suggestion
```ShellSession
$ pkgs/misc/vim-plugins/update.py -i vim-plugin-names -o generated.nix --no-commit
```
with the content of vim-plugin-names be like:
```ShellSession
$ pwntester/octo.nvim
```
```

File: doc/languages-frameworks/vim.section.md [hunk 3]
```diff
@@ -181,6 +181,22 @@ of precompiled grammars, you can use `nvim-treesitter.withPlugins` function:
...
+You can reference the generated vim plugins by:
+```nix
+myVimPlugins = pkgs.vimPlugins.extend (
+  (pkgs.callPackage generated.nix {})
```
Comment (jlesquembre) [inline doc/languages-frameworks/vim.section.md:196]: It should be
```nix
    (pkgs.callPackage ./generated.nix {
      buildVimPluginFrom2Nix = pkgs.vimUtils.buildVimPluginFrom2Nix;
    })
```

**Review by jlesquembre — CHANGES_REQUESTED** [review]

File: doc/languages-frameworks/vim.section.md (hunk 3 above)
Comment (teto) [reply doc/languages-frameworks/vim.section.md]: fixed

**Comment by teto** [comment]
Now that I've removed quite a bit of vim doc (related to VAM, the vim addon manager), I think there is space for this bit of information. Can we merge or are there still comments ?

**Comment by jlesquembre** [comment]
Looks good to me, I'd like to get it finally merged

File: doc/languages-frameworks/vim.section.md [hunk 4]
```diff
@@ -182,6 +182,23 @@ of precompiled grammars, you can use `nvim-treesitter.withPlugins` function:
 
 To enable all grammars packaged in nixpkgs, use `(pkgs.vimPlugins.nvim-treesitter.withPlugins (plugins: pkgs.tree-sitter.allGrammars))`.
 
+You can use the updater script to generate basic packages out of a custom vim
+plugin list:
+```
+pkgs/applications/editors/vim/plugins/update.py -i vim-plugin-names -o generated.nix --no-commit
+```
+with the content of "vim-plugin-names" be like:
```
Comment (fricklerhandwerk) [inline doc/languages-frameworks/vim.section.md:190]:
```suggestion
with the contents of `vim-plugin-names` being for example:
```

Also please use empty lines between paragraphs for better source readability.

**Comment by teto** [comment]
I've added a mention about vim-plug being deprecated for neovim.

**Comment by teto** [comment]
@jlesquembre @fricklerhandwerk would you mind doing another pass/merge if tha's fine ?

File: doc/languages-frameworks/vim.section.md [hunk 5]
```diff
@@ -243,10 +231,23 @@ Alternatively, set the number of processes to a lower count to avoid rate-limiti
 ./pkgs/applications/editors/vim/plugins/update.py --proc 1
 ```
 
-## Important repositories {#important-repositories}
+## How to maintain an out-of-tree overlay of vim plugins ?
+
+You can use the updater script to generate basic packages out of a custom vim
+plugin list:
+```
+pkgs/applications/editors/vim/plugins/update.py -i vim-plugin-names -o generated.nix --no-commit
```
Comment (fricklerhandwerk) [inline doc/languages-frameworks/vim.section.md:239]: Can you run this out of tree, i.e. with `nix-shell`?

File: doc/languages-frameworks/vim.section.md [hunk 6]
```diff
@@ -243,10 +231,23 @@ Alternatively, set the number of processes to a lower count to avoid rate-limiti
...
+```
+with the contents of "vim-plugin-names" being for example:
```
Comment (fricklerhandwerk) [inline doc/languages-frameworks/vim.section.md:241]:
```suggestion
with the contents of `vim-plugin-names` being for example:
```

File: nixos/doc/manual/release-notes/rl-2211.section.md [hunk 7]
```diff
@@ -148,7 +148,8 @@ Available as [services.patroni](options.html#opt-services.patroni.enable).
 - The `meta.mainProgram` attribute of packages in `wineWowPackages` now defaults to `"wine64"`.
 
 - (Neo)Vim can not be configured with `configure.pathogen` anymore to reduce maintainance burden.
-Use `configure.packages` instead.
+Use `configure.packages` instead. Neovim can not be configured with plug anymore
+too (still works for vim).
```
Comment (fricklerhandwerk) [inline nixos/doc/manual/release-notes/rl-2211.section.md:152]:
```suggestion
  Use `configure.packages` instead.
- Neovim can not be configured with plug anymore (still works for vim).
```

**Review by fricklerhandwerk — CHANGES_REQUESTED** [review]
This is a great addition, thank you. A few minor comments on presentation, other than that the explanations are very helpful. Please add newlines around markdown code blocks, this increases readability of the source.

A general thing which I would like to see, but would not block on:
Could we establish a more consistent naming of tools? Sometimes it is "Vim" or "Neovim", sometimes it is "vim" or "neovim".

On https://neovim.io they address it as "Vim" and "Neovim", and I think we should stick to that.

I know this is inconsistent in the context and in many areas of the manual, but we can only reasonably improve this one PR at a time.

File: doc/languages-frameworks/vim.section.md (hunk 5 above)
Comment (teto) [reply doc/languages-frameworks/vim.section.md:240]: That would be a logical followup indeed. Not sure how to expose that yet so I'll postponethe suggestion.

**Comment by teto** [comment]
@fricklerhandwerk  I've used Neovim everywhere as you mentioned and incorporated your good (=all) suggestions :) . Merging.

**Pull request merged**